libpod/container_internal.go

```diff
@@ -845,6 +846,15 @@ func (c *Container) cleanup() error {
 		}
 	}
 
+	// Remove the container from the runtime, if necessary
+	if err := c.cleanupRuntime(ctx); err != nil {
+		if lastError != nil {
+			logrus.Errorf("Error removing container %s from OCI runtime: %v", c.ID(), err)
+		} else {
```

Comment:
This should not be an else here.  You wan to always set lastError=err.

Reply:
The way this one is set up, we print the new error, and return lastError. I can rework it if you want, but I'd also have to change the other one here.

Reply:
Ok I missed that.  So you want to return the first error that happens. SGTM